data_manager_select_index_by_path/data_manager/path_name_value_key_manager.py

```diff
@@ -105,7 +116,7 @@ def get_index_properties(self) -> dict:
             raise ValueError(
                 "'{0}' not a supported table name".format(
                     self.data_table_name))
-        return index_properties
+        return indexes_schema().validate(index_properties)
```

Comment:
This validates the schema after using (part) of the supplied yaml.  
I think it would be possible to construct the schema based on `self.data_table_name`, such that all required information is validated, no?

Reply:
Yes, but this requires some advanced nested schema stuff.
Could you explain to me how that works? I don't mind changing the library to python-jsonschema.

Reply:
No need to make it extremely complicated. 

Something like this would work:

```python
def indexes_schema(table_name):
    return Schema({table_name: { 
        "name": str,
        Optional('prefix'): bool,
        ....  # the other values
    }}, ignore_extra_keys=True)
```

`get_index_properties` can then be very simple:

```python
def get_index_properties(self) -> dict:
     with self.indexes_properties_file.open('r') as properties_file:
         indexes = yaml.safe_load(properties_file)
     return indexes_schema(self.data_table_name).validate(indexes)[self.data_table.name]  # keyerror should never occur, since validation happens in schema)
```

Reply:
This destroys all information about types in the schema...